### ISSUE TITLE
Simplified `Item->addEnchantment()` API

### ISSUE THREAD

**Issue by dktapps** [pmmp/PocketMine-MP#5971]
## Justification
The word `enchantment` is obnoxiously long, yet grammatically more correct in this situation than `enchant`, which is a verb.

However, this is undeniably nasty to read and write:
```php
$item->addEnchantment(new EnchantmentInstance(VanillaEnchantments::SHARPNESS(), 5));
```

It used to be worse in PM3 before the introduction of `VanillaEnchantments`.

The case for getting rid of `EnchantmentInstance` was discussed in #3065, and it was decided not to remove it. However, that doesn't mean that users of `addEnchantment()` (the most commonly used API) should be forced to write `new EnchantmentInstance` every time they want to enchant an item.

A simpler `addEnchantment(Enchantment $enchantment, int $level)` would make this much less obnoxious at the price of a BC break, but I don't think anyone would be unhappy about such a change, reducing the above to:
```php
$item->addEnchantment(VanillaEnchantments::SHARPNESS(), 5);
```

It's also possible we could introduce a second method with this simplified signature, but it's hard to think of a good name that wouldn't be significantly longer than the existing one, negating its benefit. Therefore, unless someone can come up with an excellent name for it, this change proposal will target PM6.

**Comment by nicholass003** [comment]
i have a recommendation for this, how about adding a new method called `enchant`? without needing to remove or replace the addEnchantment method, so here there will be no BC breaks.

`ItemEnchantmentHandlingTrait.php`
```php
public function enchant(Enchantment $enchantment, int $level = 1) : void{
	$this->addEnchantment(new EnchantmentInstance($enchantment, $level);
}
```
something like :
```php
$item->enchant(VanillaEnchantments::SHARPNESS(), 5);
```
grammatically, `enchant` is a verb, so in my opinion this is quite precise and short, if in the **Justification** it is explained that `enchant` means `addEnchant` and `addEnchantment` is the right choice, then `enchant` here is short for `addEnchantment`.

**Comment by dktapps** [comment]
That's not a bad idea tbh. My only concern is that people might get mixed up between the two methods.

**Comment by nicholass003** [comment]
> That's not a bad idea tbh. My only concern is that people might get mixed up between the two methods.

that's true, because people are so used to using `addEnchantment` that if they don't read the changelog they definitely won't know about the existence of the enchant method.
so that way i also have one more option, but this is probably a bad idea, by including the `Enchantment::class` option in the $enchantment parameter in the `addEnchantment` method

changes to `addEnchantment` method :
```php
	/**
	 * @return $this
	 */
	public function addEnchantment(EnchantmentInstance|Enchantment $enchantment, ?int $level = null) : self{
		if($enchantment instanceof Enchantment && is_int($level)){
			$enchantment = new EnchantmentInstance($enchantment, $level);
		}
		$this->enchantments[spl_object_id($enchantment->getType())] = $enchantment;
		return $this;
	}
```
